rust/src/pgsql/pgsql.rs

```diff
@@ -341,16 +356,27 @@ impl PgsqlState {
             );
             match PgsqlState::state_based_req_parsing(self.state_progress, start) {
                 Ok((rem, request)) => {
-                    sc_app_layer_parser_trigger_raw_stream_reassembly(flow, Direction::ToServer as i32);
                     start = rem;
-                    if let Some(state) = PgsqlState::request_next_state(&request) {
+                    let temp_state: PgsqlStateProgress;
```

Comment:
this could be made slightly more compact by using the same tuple logic as with the response?